### PR TITLE
feat: blast contract deployments

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

Potential UMA integrations would need to use Optimistic Oracle on Blast network.

**Summary**

Adds Optimistic Oracle contract deployments for Blast network.

**Details**

Changes proposed in this PR:
- Add Blast network as a supported network
- Deploy Blast contracts and L1 Parent Messenger
- Verify contracts
- Run all the required L1 and L2 configurations
- Update configuration and verification scripts (including new default gas limit of 500k in parent messenger)
- Update readme

This requires governance action set Blast parent messenger on L1 (see verification script below):

```
$ yarn hardhat verify-xchain --network mainnet --l2 blast

🌕 Verifying L1 contract state 🌕
  GovernorHub
    - Owner set to Governor: ✅
    - Messenger for chain ID 81457 set to ParentMessenger: ❌ ==> Requires governance action
  OracleHub
    - Owner set to Governor: ✅
    - Messenger for chain ID 81457 set to ParentMessenger: ❌ ==> Requires governance action
  Blast_ParentMessenger
    - Owner set to Governor: ✅
    - Child chain ID set to 81457: ✅
    - Set childMessenger address: ✅
    - Set oracleHub address: ✅
    - Set governorHub address: ✅
    - Set oracleSpoke address: ✅
    - Set governorSpoke address: ✅
  Registry
    - OracleHub registered: ✅

🌚 Verifying L2 contract state 🌚
  Registry
    - OptimisticOracle registered: ✅
    - OptimisticOracleV2 registered: ✅
    - OptimisticOracleV3 registered: ✅
    - Owned by GovernorSpoke: ✅
  Store
    - Owned by GovernorSpoke: ✅
  IdentifierWhitelist
    - Owned by GovernorSpoke: ✅
  OptimisticOracleV3
    - Owned by GovernorSpoke: ✅
  AddressWhitelist
    - Owned by GovernorSpoke: ✅
  Blast_ChildMessenger
    - Set oracleSpoke: ✅
    - Set parentMessenger: ✅
  Finder
    - Owned by GovernorSpoke: ✅
    - Set "Store" in Finder: ✅
    - Set "IdentifierWhitelist": ✅
    - Set "CollateralWhitelist" in Finder: ✅
    - Set "Oracle" in Finder (to OracleSpoke): ✅
    - Set "OptimisticOracle" in Finder: ✅
    - Set "OptimisticOracleV2" in Finder: ✅
    - Set "OptimisticOracleV3" in Finder: ✅
    - Set "ChildMessenger" in Finder: ✅
```

Following collateral currencies have been pre-approved on Blast:

- WETH at [0x4300000000000000000000000000000000000004]([0x4300000000000000000000000000000000000004](https://blastscan.io/address/0x4300000000000000000000000000000000000004)) with 0.135 final fee. This is rebasing token that captures yield from staking ETH on L1 bridge.
- USDB at [0x4300000000000000000000000000000000000003](https://blastscan.io/address/0x4300000000000000000000000000000000000003) with 250 final fee. Users who bridge stablecoins receive USDB, Blast’s auto-rebasing stablecoin. The yield for USDB comes from MakerDAO’s on-chain T-Bill protocol. USDB can be redeemed for DAI when bridging back to Ethereum.
- WBTC at [0xF7bc58b8D8f97ADC129cfC4c9f45Ce3C0E1D2692](https://blastscan.io/address/0xF7bc58b8D8f97ADC129cfC4c9f45Ce3C0E1D2692) with 0.025 final fee.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

**Issue(s)**

Fixes https://linear.app/uma/issue/UMA-2515/blast-contract-deployments

### CODE REVIEW

**Review by md0x — APPROVED** [review]
Looks good to me. After reading about the rebasing tokens in Blast, I agree that the rebases are only incremental; thus, they won't alter the functioning of the optimistic oracles and other related contracts. Also, we don't use `erc20.balanceOf()` in any relevant logic of our contracts.

File: packages/common/src/hardhat/tasks/xchainSetup.ts [hunk 1]
```diff
@@ -292,6 +309,13 @@ task("setup-l1-optimism-xchain", "Configures L1 cross chain smart contracts for
   }
 );
 
+task("setup-l1-blast-xchain", "Configures L1 cross chain smart contracts for Blast bridge").setAction(async function (
```
Comment (mrice32) [inline packages/common/src/hardhat/tasks/xchainSetup.ts:312]: Don't have to do this in this PR, but maybe it makes sense to start making this stuff as generic as possible.

In this case, I think having a generic task for OP Stack chains that just takes a chainId arg. Something like `yarn hardhat setup-l1-op-xchain --chainId <chainId>`.

Thoughts?

**Review by mrice32 — APPROVED** [review]
LGTM!

File: packages/common/src/hardhat/tasks/xchainSetup.ts (hunk 1 above)
Comment (Reinis-FRP) [reply packages/common/src/hardhat/tasks/xchainSetup.ts:312]: agree, should generalize when adding new OS stack chains

**Pull request merged**